evm/precompiles/ecpairing.py

```diff
@@ -0,0 +1,107 @@
+from cytoolz import (
+    curry,
+    pipe,
+)
+from py_ecc import (
+    optimized_bn128 as bn128,
+)
+
+from evm import constants
+from evm.exceptions import (
+    ValidationError,
+    VMError,
+)
+from evm.utils.bn128 import (
+    validate_point,
+)
+from evm.utils.numeric import (
+    big_endian_to_int,
+)
+from evm.utils.padding import (
+    pad32,
+)
+
+
+ZERO = (bn128.FQ2.one(), bn128.FQ2.one(), bn128.FQ2.zero())
```

Comment:
I find this an odd name choice for a 3-tuple with seemingly non-zero items, but maybe I'm missing something?

Reply:
This code is a refactored version of vitalik's implementation in pyethereum.  I'm not familiar enough with what is *actually* going on here to know what a better name would be :(